Get rid of panic in stats/ fix dir lxc

jessfraz opened the pull request.

jessfraz committed:
Get rid of panic in stats for lxc

Fix containers dir

Docker-DCO-1.1-Signed-off-by: Jessie Frazelle <jess@docker.com> (github: jfrazelle)

jfrazelle commented:
ping @crosbymichael

jfrazelle commented:
ugh wait testing more

jfrazelle commented:
ok ya we good

LK4D4 commented on daemon/daemon.go:
```diff
@@ -913,7 +913,8 @@ func NewDaemonFromDirectory(config *Config, eng *engine.Engine) (*Daemon, error)
 
-	daemonRepo := path.Join(config.Root, "containers")
+	const runDir = "/var/run/docker"
+	daemonRepo := path.Join(runDir, "containers")
```
But that's not quite right I think. That data should survive restart.
ping @crosbymichael

jfrazelle replied:
problem was on lxc it was frekaing out about not being able to find /var/run/docker/containers, so maybe its referencing that somewhere else will look

jfrazelle replied:
ah hang on i got it

jfrazelle replied:
its line 702,

LK4D4 commented on daemon/execdriver/lxc/driver.go:
```diff
@@ -783,5 +783,8 @@ func (d *driver) Exec(c *execdriver.Command, processConfig *execdriver.ProcessCo
 
 func (d *driver) Stats(id string) (*execdriver.ResourceStats, error) {
+	if _, ok := d.activeContainers[id]; !ok {
+		return &execdriver.ResourceStats{}, fmt.Errorf("%s is not a key in active containers", id)
```
better return nil instead.

jfrazelle commented:
updated its not pretty

jfrazelle commented:
boom got it, gitblame lead to https://github.com/docker/docker/commit/8dc5791f7302aea0734a26dc1f54d96b04771b8f which makes sense because the panic was **NOT** fixed in lxc omg my life i literally cant even ping @LK4D4 @crosbymichael

tiborvass commented:
LGTM

jfrazelle commented:
Ah man 1 stupid test on lxc failed I'll fix in a bit

jfrazelle commented:
it's unrelated tho

jessfraz committed:
fix 2 integration tests on lxc

Docker-DCO-1.1-Signed-off-by: Jessica Frazelle <jess@docker.com> (github: jfrazelle)

duglin commented:
LGTM
